Makefile

```diff
@@ -13,29 +13,43 @@ create-env:
 
 .PHONY: install
 install:
-	$(PIP) install --upgrade $(PIP)
+	$(PIP) install --upgrade pip
+	$(PIP) install .
 	$(PIP) install -r $(TEST_DIR)/utils/requirements.txt
+	$(PIP) install pre-commit pytest pytest-cov codecov
 
 .PHONY: test
 test: .cleanCoverage
-	$(PYTEST) $(TEST_DIR)/unit
+	$(PYTEST) $(TEST_DIR)/unit -v
 
 .cleanCoverage:
 	@echo 'cleaning coverage files ...'
-	rm -f .coverage
-	rm -fr htmlcov/
+	rm -rf .coverage htmlcov/
 
 .PHONY: open-report
 open-report: .cleanCoverage
 	$(PYTEST) --cov=$(YARPC) --cov-report=term-missing --cov-report=html
 	open htmlcov/index.html
 
 .PHONY: ci-test
-ci-test:
-	$(PIP) install -r tests/utils/requirements.txt
-	pre-commit run --all-files
-	$(PIP) install .
-	$(PYTEST) --cov --cov-report=xml
+ci-test: lint
+	$(PYTEST) --cov --cov-report=xml -v
+
+.PHONY: flake8
+flake8:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec flake8 "{}" +
+
+.PHONY: black
+black:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec black "{}" +
+
+.PHONY: mypy
+mypy:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec mypy --ignore-missing-imports "{}" +
```

Comment:
I don't think `--ignore-missing-imports` is needed, it is a workaround for pre-commit

Reply:
if I remove that there is an error in the `setup.py` file that comes up.
![image](https://user-images.githubusercontent.com/36493537/66707672-2b1c9b00-ed12-11e9-885d-7d5ee135f007.png)

Reply:
Yeah, that's the only problem currently. I didn't add types to setup.py. `flake8 $(SOURSES)` should work

Reply:
That is not going to work because the way you run `mypy` is passing a python file not a folder, and that is why I use `find` to gather all the python file to run the recursively with `mypy`

Reply:
All these tools support using directory name / list of directory names as input. Do you get errors running it this way?

Reply:
```
.PHONY: mypy
mypy:
  mypy yarpc examples
```

What output do you get by running this?

Reply:
Ooh okay I see what you mean, you just want to pass the directories that you want to check, so this would be the same for all `flake8`, `mypy`, `isort`, and `black` you only want to check `yarpc` and `examples`?

Reply:
Yes, should be the same for all these tools